staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -92,7 +137,12 @@ func WithMaxInFlightLimit(
 
 			select {
 			case c <- true:
-				defer func() { <-c }()
+				nonMutatingLen := len(nonMutatingChan)
```

Comment:
should probably only update the length of the channel we're using in this request

Reply:
Done.